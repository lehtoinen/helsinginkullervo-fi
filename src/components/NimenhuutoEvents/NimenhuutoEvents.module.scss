@import '../../assets/scss/colors';

.root {
  position: relative;
  display: flow-root;
  font-size: 0.8em;
  margin: 1em;

  .tab {
    display: none;
    position: absolute;
    box-shadow: 1em 1.5em $color-base-dark;
    margin: {
      right: 0.5em;
      bottom: 2em;
    }

    &:before {
      display: block;
      position: absolute;
      left: -1px;
      top: -1px;
      width: calc(100% + 2px);
      height: calc(100% + 2px);
      content: "";
      background: $color-base-light;
      background: $color-complimentary;
      z-index: -1;
    }

    &.active {
      display: flow-root;
    }
  }
}