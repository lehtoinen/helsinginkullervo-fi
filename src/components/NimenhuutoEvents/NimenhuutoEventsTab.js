import React from 'react';
import PropTypes from 'prop-types';
import CSV from 'comma-separated-values';
import moment from 'moment';

import styles from './NimenhuutoEventsTab.module.scss';

const NimenhuutoEventsTab = ({ label, items, loading, error }) => {
  let state;
  if (error) {
    return <div className="error">Lataaminen epäonnistui!</div>;
  } else if (loading && !items.length) {
    return <div className="loading">Ladataan...</div>;
  } else if (!loading && !items.length) {
    return <div className="empty">Ei tulevia tapahtumia.</div>;
  }

  return (
    <div className={styles.root}>
      {items.map(item => {
        const { subject, location, date, time, url, description } = item;
        return (
          <a
            href={url}
            target="_blank"
            rel="noopener noreferrer"
            className={`event-row ${styles.row}`}
            key={`${subject}${date}${time}`}
          >
            <DateBadge dateStr={date} />
            <div className={styles.info}>
              <div className={styles.subject}>
                {/* subject contains the name of the nimenhuuto-instance (for example "Kullervo: ...")
                stripping that to avoid unnecessary repetition */
                subject.substr(subject.indexOf(': ') + 2)}
                {/* {description && <span>&nbsp;i</span>} */}
              </div>
              <div className={styles.details}>
                <span className={styles.time}>Klo {time}</span>
                <span className={styles.location}>{location}</span>
              </div>
            </div>
          </a>
        );
      })}
    </div>
  );
};

NimenhuutoEventsTab.propTypes = {
  label: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.bool.isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      subject: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      date: PropTypes.date,
      time: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      description: PropTypes.string,
    })
  ).isRequired,
};

const DateBadge = ({ dateStr }) => {
  moment.locale('fi');
  const eventDate = moment(dateStr);
  const weekday = eventDate.format('dd');
  const date = eventDate.format('DD.MM');

  return (
    <div className={styles.day}>
      <div className={styles.weekday}>{weekday}</div>
      <div className={styles.date}>{date}</div>
    </div>
  );
};

export default NimenhuutoEventsTab;

// TODO/IMPROVEMENT: Add a collapse for the description when available  (> / ▼ -icon)
