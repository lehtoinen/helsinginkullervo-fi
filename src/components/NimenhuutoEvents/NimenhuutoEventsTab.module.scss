@import '../../assets/scss/colors';

.row {
  padding: 0.65em calc(0.65em + 0.45em);
  text-decoration: none;
  display: flex;
  background: $color-white;
  color: $color-black;

  &:not(:first-child) {
    border-top: 1px solid $color-complimentary;
  }

  &:hover {
    border-left: 0.45em solid $color-base-mid;
    padding-left: 0.65em;
    opacity: 1;
  }

  .info {
    display: flex;
    justify-content: center;
    flex-direction: column;
    text-align: left;
    line-height: 1.35;
  }

  .subject {
    font-size: 1em;
    margin-bottom: 0.115em;
    color: $color-base-mid;
    
    &::first-letter {
      text-transform: uppercase;
    }
  }

  .time {
    margin-right: 0.35em;
  }

  .details {
    font-size: 0.85em;
  }

  .location {
    color: $color-grey;
  }

  .day {
    text-align: center;
    margin-right: 0.85em;
    line-height: 0.975;

    .weekday {
      font-size: 1.65em;
      font-weight: bold;
      text-transform: uppercase;
      color: $color-base-dark;
    }

    .date {
      font-size: 0.85em;
      color: $color-grey;
    }
  }
}