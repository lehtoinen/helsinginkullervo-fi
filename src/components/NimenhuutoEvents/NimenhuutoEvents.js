import React from 'react';
import PropTypes from 'prop-types';
import CSV from 'comma-separated-values';

import NimenhuutoEventsTab from './NimenhuutoEventsTab';

import styles from './NimenhuutoEvents.module.scss';

class NimenhuutoEvents extends React.Component {
  constructor(props) {
    super();
    // hydrate state here if we have something stored!

    let sources = {};

    props.sources.forEach(source => {
      const { label, url } = source;
      sources[url] = { label, url, fetching: true, error: false, items: [] };
    });

    this.state = { sources, count: props.count };
  }

  componentDidMount() {
    const { sources } = this.props;
    sources.forEach(source => this.fetchEvents(source));
  }

  fetchEvents(source) {
    fetch(`${source.url}/calendar/csv`)
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }
        return response.text();
      })
      .then(data => {
        if (!data) throw Error('No data');
        const csv = new CSV(data, { header: true });
        const parsed = csv.parse();
        const items = parsed.map(item => {
          // the Description field has the url to the event in the beginning + description text if given (space-separated)
          const ind = item.Description.indexOf(' ');
          const url =
            ind > -1 ? item.Description.substring(0, ind) : item.Description;
          const description = ind > -1 ? item.Description.substr(ind) : null;

          return {
            subject: item.Subject,
            location: item.Location,
            date: new Date(item['Start Date']),
            time: item['Start Time'],
            url,
            description,
          };
        });

        // copy current state, don't mutate!
        const sources = { ...this.state.sources };

        sources[source.url] = {
          ...sources[source.url],
          ...{ fetching: false, items },
        };
        this.setState({ sources });
      })
      .catch(e => {
        const sources = { ...this.state.sources };

        sources[source.url] = {
          ...sources[source.url],
          ...{ fetching: false, error: true },
        };

        this.setState({ sources });
      });
  }

  render() {
    const { sources, count } = this.state;

    return (
      <div className={styles.root}>
        {Object.keys(sources).map((key, index) => {
          const source = sources[key];
          return (
            <div
              key={key}
              className={`${styles.tab} ${
                styles[index === 0 ? 'active' : 'inactive']
              }`}
            >
              <NimenhuutoEventsTab
                label={source.label}
                items={source.items ? source.items.slice(0, count) : null}
                loading={source.fetching}
                error={source.error}
              />
            </div>
          );
        })}
      </div>
    );
  }
}

NimenhuutoEvents.propTypes = {
  // data sources for the events
  sources: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })
  ).isRequired,
  // amount of events to show on load
  count: PropTypes.number,
};

export default NimenhuutoEvents;
