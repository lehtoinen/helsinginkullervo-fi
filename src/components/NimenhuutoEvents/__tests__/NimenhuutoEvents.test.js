import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NimenhuutoEvents from '../NimenhuutoEvents';
import NimenhuutoEventsTab from '../NimenhuutoEventsTab';

import mockData from '../__mocks__/nimenhuutoMock.js';

configure({ adapter: new Adapter() });
global.fetch = require('jest-fetch-mock');

const sources = [{ label: 'Foo', url: 'foo' }, { label: 'Bar', url: 'bar' }];

function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}

describe('NimenhuutoEvents', () => {
  test('fetches data for all sources', () => {
    fetch.mockResponse(mockData, { status: 200 });
    const component = mount(<NimenhuutoEvents sources={sources} />);

    return flushPromises().then(() => {
      expect(fetch.mock.calls).toHaveLength(2);
    });
  });

  test('renders with valid sources', () => {
    fetch.mockResponse(mockData, { status: 200 });
    const component = mount(<NimenhuutoEvents sources={sources} />);

    return flushPromises().then(() => {
      component.update();
      expect(component).toMatchSnapshot();
    });
  });

  test('renders with invalid csv data', () => {
    fetch.mockResponse('foobar', { status: 200 });
    const component = mount(<NimenhuutoEvents sources={sources} />);

    return flushPromises().then(() => {
      component.update();
      expect(component).toMatchSnapshot();
    });
  });

  test('renders with erroneous http response for csv data fetch', () => {
    fetch.mockResponse('foobar', { status: 404 });
    const component = mount(<NimenhuutoEvents sources={sources} />);

    return flushPromises().then(() => {
      component.update();
      expect(component).toMatchSnapshot();
    });
  });
});
