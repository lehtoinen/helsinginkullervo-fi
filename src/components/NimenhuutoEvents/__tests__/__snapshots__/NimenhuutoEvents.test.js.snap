// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NimenhuutoEvents renders with erroneous http response for csv data fetch 1`] = `
<NimenhuutoEvents
  sources={
    Array [
      Object {
        "label": "Foo",
        "url": "foo",
      },
      Object {
        "label": "Bar",
        "url": "bar",
      },
    ]
  }
>
  <div
    className="root"
  >
    <div
      className="tab active"
      key="foo"
    >
      <NimenhuutoEventsTab
        error={true}
        items={Array []}
        label="Foo"
        loading={false}
      >
        <div
          className="error"
        >
          Lataaminen epäonnistui!
        </div>
      </NimenhuutoEventsTab>
    </div>
    <div
      className="tab inactive"
      key="bar"
    >
      <NimenhuutoEventsTab
        error={true}
        items={Array []}
        label="Bar"
        loading={false}
      >
        <div
          className="error"
        >
          Lataaminen epäonnistui!
        </div>
      </NimenhuutoEventsTab>
    </div>
  </div>
</NimenhuutoEvents>
`;

exports[`NimenhuutoEvents renders with invalid csv data 1`] = `
<NimenhuutoEvents
  sources={
    Array [
      Object {
        "label": "Foo",
        "url": "foo",
      },
      Object {
        "label": "Bar",
        "url": "bar",
      },
    ]
  }
>
  <div
    className="root"
  >
    <div
      className="tab active"
      key="foo"
    >
      <NimenhuutoEventsTab
        error={false}
        items={Array []}
        label="Foo"
        loading={false}
      >
        <div
          className="empty"
        >
          Ei tulevia tapahtumia.
        </div>
      </NimenhuutoEventsTab>
    </div>
    <div
      className="tab inactive"
      key="bar"
    >
      <NimenhuutoEventsTab
        error={false}
        items={Array []}
        label="Bar"
        loading={false}
      >
        <div
          className="empty"
        >
          Ei tulevia tapahtumia.
        </div>
      </NimenhuutoEventsTab>
    </div>
  </div>
</NimenhuutoEvents>
`;

exports[`NimenhuutoEvents renders with valid sources 1`] = `
<NimenhuutoEvents
  sources={
    Array [
      Object {
        "label": "Foo",
        "url": "foo",
      },
      Object {
        "label": "Bar",
        "url": "bar",
      },
    ]
  }
>
  <div
    className="root"
  >
    <div
      className="tab active"
      key="foo"
    >
      <NimenhuutoEventsTab
        error={false}
        items={
          Array [
            Object {
              "date": 2018-03-23T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547347",
            },
            Object {
              "date": 2018-03-25T00:00:00.000Z,
              "description": "  
Ulkotreenit Pirkkolan tekiksellä. 1 h, puoli kenttää (täysi jos on vapaata).",
              "location": "Pirkkola TN, Pirkkolan urheilupuisto, Helsinki",
              "subject": "Kullervo: Treenit",
              "time": "20:00",
              "url": "https://kullervo.nimenhuuto.com/events/10469058",
            },
            Object {
              "date": 2018-04-06T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547349",
            },
            Object {
              "date": 2018-04-13T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547350",
            },
            Object {
              "date": 2018-04-20T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547351",
            },
            Object {
              "date": 2018-04-27T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547352",
            },
            Object {
              "date": 2018-04-28T00:00:00.000Z,
              "description": "  
Perinteinen Herrasmiesturnaus 7v7, 
4 peliä ja 25 min/peli, 
13 ensimmäistä mukaan ja loput varalle.",
              "location": "Väinämöinen TN, Väinämöisenkatu 13 00100 Helsinki",
              "subject": "Kullervo: Herrasmiesturnaus Ikiksille",
              "time": "15:30",
              "url": "https://kullervo.nimenhuuto.com/events/10693998",
            },
          ]
        }
        label="Foo"
        loading={false}
      >
        <div
          className="root"
        >
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547347"
            key="Kullervo: karvapalloa varttuneilleFri Mar 23 2018 02:00:00 GMT+0200 (EET)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-03-23T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  23.03
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10469058"
            key="Kullervo: TreenitSun Mar 25 2018 02:00:00 GMT+0200 (EET)20:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-03-25T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  su
                </div>
                <div
                  className="date"
                >
                  25.03
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                Treenit
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  20:00
                </span>
                <span
                  className="location"
                >
                  Pirkkola TN, Pirkkolan urheilupuisto, Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547349"
            key="Kullervo: karvapalloa varttuneilleFri Apr 06 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-06T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  06.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547350"
            key="Kullervo: karvapalloa varttuneilleFri Apr 13 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-13T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  13.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547351"
            key="Kullervo: karvapalloa varttuneilleFri Apr 20 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-20T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  20.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547352"
            key="Kullervo: karvapalloa varttuneilleFri Apr 27 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-27T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  27.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10693998"
            key="Kullervo: Herrasmiesturnaus IkiksilleSat Apr 28 2018 03:00:00 GMT+0300 (EEST)15:30"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-28T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  la
                </div>
                <div
                  className="date"
                >
                  28.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                Herrasmiesturnaus Ikiksille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  15:30
                </span>
                <span
                  className="location"
                >
                  Väinämöinen TN, Väinämöisenkatu 13 00100 Helsinki
                </span>
              </div>
            </div>
          </a>
        </div>
      </NimenhuutoEventsTab>
    </div>
    <div
      className="tab inactive"
      key="bar"
    >
      <NimenhuutoEventsTab
        error={false}
        items={
          Array [
            Object {
              "date": 2018-03-23T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547347",
            },
            Object {
              "date": 2018-03-25T00:00:00.000Z,
              "description": "  
Ulkotreenit Pirkkolan tekiksellä. 1 h, puoli kenttää (täysi jos on vapaata).",
              "location": "Pirkkola TN, Pirkkolan urheilupuisto, Helsinki",
              "subject": "Kullervo: Treenit",
              "time": "20:00",
              "url": "https://kullervo.nimenhuuto.com/events/10469058",
            },
            Object {
              "date": 2018-04-06T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547349",
            },
            Object {
              "date": 2018-04-13T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547350",
            },
            Object {
              "date": 2018-04-20T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547351",
            },
            Object {
              "date": 2018-04-27T00:00:00.000Z,
              "description": null,
              "location": "Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki",
              "subject": "Kullervo: karvapalloa varttuneille",
              "time": "19:00",
              "url": "https://kullervo.nimenhuuto.com/events/10547352",
            },
            Object {
              "date": 2018-04-28T00:00:00.000Z,
              "description": "  
Perinteinen Herrasmiesturnaus 7v7, 
4 peliä ja 25 min/peli, 
13 ensimmäistä mukaan ja loput varalle.",
              "location": "Väinämöinen TN, Väinämöisenkatu 13 00100 Helsinki",
              "subject": "Kullervo: Herrasmiesturnaus Ikiksille",
              "time": "15:30",
              "url": "https://kullervo.nimenhuuto.com/events/10693998",
            },
          ]
        }
        label="Bar"
        loading={false}
      >
        <div
          className="root"
        >
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547347"
            key="Kullervo: karvapalloa varttuneilleFri Mar 23 2018 02:00:00 GMT+0200 (EET)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-03-23T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  23.03
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10469058"
            key="Kullervo: TreenitSun Mar 25 2018 02:00:00 GMT+0200 (EET)20:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-03-25T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  su
                </div>
                <div
                  className="date"
                >
                  25.03
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                Treenit
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  20:00
                </span>
                <span
                  className="location"
                >
                  Pirkkola TN, Pirkkolan urheilupuisto, Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547349"
            key="Kullervo: karvapalloa varttuneilleFri Apr 06 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-06T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  06.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547350"
            key="Kullervo: karvapalloa varttuneilleFri Apr 13 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-13T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  13.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547351"
            key="Kullervo: karvapalloa varttuneilleFri Apr 20 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-20T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  20.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10547352"
            key="Kullervo: karvapalloa varttuneilleFri Apr 27 2018 03:00:00 GMT+0300 (EEST)19:00"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-27T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  pe
                </div>
                <div
                  className="date"
                >
                  27.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                karvapalloa varttuneille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  19:00
                </span>
                <span
                  className="location"
                >
                  Helsingin Ranskalais-Suomalainen Koulu, Raumantie 4, 00350 Helsinki
                </span>
              </div>
            </div>
          </a>
          <a
            className="event-row row"
            href="https://kullervo.nimenhuuto.com/events/10693998"
            key="Kullervo: Herrasmiesturnaus IkiksilleSat Apr 28 2018 03:00:00 GMT+0300 (EEST)15:30"
            rel="noopener noreferrer"
            target="_blank"
          >
            <DateBadge
              dateStr={2018-04-28T00:00:00.000Z}
            >
              <div
                className="day"
              >
                <div
                  className="weekday"
                >
                  la
                </div>
                <div
                  className="date"
                >
                  28.04
                </div>
              </div>
            </DateBadge>
            <div
              className="info"
            >
              <div
                className="subject"
              >
                Herrasmiesturnaus Ikiksille
              </div>
              <div
                className="details"
              >
                <span
                  className="time"
                >
                  Klo 
                  15:30
                </span>
                <span
                  className="location"
                >
                  Väinämöinen TN, Väinämöisenkatu 13 00100 Helsinki
                </span>
              </div>
            </div>
          </a>
        </div>
      </NimenhuutoEventsTab>
    </div>
  </div>
</NimenhuutoEvents>
`;
